src/restruct_poc/solver/constraints.hpp

```diff
@@ -29,19 +26,21 @@ struct Constraints {
         double axis_x[3];          // Unit vector representing x rotation axis
         double axis_y[3];          // Unit vector representing y rotation axis
         double axis_z[3];          // Unit vector representing z rotation axis
+        double* control;           //< Control signal
```

Comment:
Delete this variable.  It can't work on GPU (a pointer to host memory can't be used) and is only used as a flag (it doesn't look like it's ever set)